engine/collection/ingest/rate_limiter.go

```diff
@@ -0,0 +1,74 @@
+package ingest
+
+import (
+	"sync"
+
+	"golang.org/x/time/rate"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type AddressRateLimiter struct {
+	mu       sync.RWMutex
+	limiters map[flow.Address]*rate.Limiter
+	limit    rate.Limit
+	burst    int // X messages allowed at one time
+}
+
+// AddressRateLimiter limits the rate of messages sent to a given address
+// It allows the given "limit" amount messages per second with a "burst" amount of messages to be sent at once
```

Comment:
@kc1116 could you review the rate limiter implementation? Please check if it's implementing the correct behavior explained here.

Reply:
Looks good this is the expected behavior, you can also cross reference with the [godoc](https://pkg.go.dev/golang.org/x/time/rate#Limiter) for the rate package